mobly/controllers/android_device_lib/snippet_client.py

```diff
@@ -78,25 +80,27 @@ def check_app_installed(self):
         """Overrides superclass."""
         # Check that the Mobly Snippet app is installed.
         if not self._adb_grep_wrapper(
-            'pm list package | grep ^package:%s$' % self.package):
+                'pm list package | grep "^package:%s[^M]\?$"' % self.package):
```

Comment:
It's literally a caret followed by a capital M. You can look at your source file in a hex editor. 5E 4D.